### PR TITLE
Identify ingress by generational node id

### PR DESCRIPTION
**Pull request by AhmedSoliman**
Identify ingress by generational node id

Leverage generational node ids to identify the ingress node instead of the socket address.

---
Stack created with [Sapling](https://sapling-scm.com). Best reviewed with [ReviewStack](https://reviewstack.dev/restatedev/restate/pull/1166).
* #1172
* #1171
* #1168
* __->__ #1166
* #1164

### CODE REVIEW

File: crates/storage-proto/proto/dev/restate/storage/v1/domain.proto [hunk 1]
```diff
@@ -336,8 +336,9 @@ message OutboxMessage {
 
     message OutboxIngressResponse {
         FullInvocationId full_invocation_id = 1;
-        SocketAddr socket_addr = 2;
+        /*SocketAddr socket_addr = 2;*/
```
Comment (tillrohrmann) [inline crates/storage-proto/proto/dev/restate/storage/v1/domain.proto:339]: Should we mark `2` as reserved (https://protobuf.dev/programming-guides/proto3/#fieldreserved)?

**Review by tillrohrmann — APPROVED** [review]
LGTM. +1 for merging.

File: crates/storage-proto/proto/dev/restate/storage/v1/domain.proto (hunk 1 above)
Comment (AhmedSoliman) [reply crates/storage-proto/proto/dev/restate/storage/v1/domain.proto:339]: I'll do that, but we are backward incompatible anyway at the moment :(

Comment (tillrohrmann) [reply crates/storage-proto/proto/dev/restate/storage/v1/domain.proto:339]: True. Then I'd also be fine with reusing `2` (given that we require a wipe anyway).

**Pull request merged**